.github/workflows/build-test-publish.yml

```diff
@@ -0,0 +1,161 @@
+name: Build, Test, Publish
+
+on:
+  pull_request:
+  push:
+    branches: ["main"]
+  schedule:
+    - cron: '0 0 * * *'
```

Comment:
I wonder if longer term we should consider making the cron less frequent at some point, particularly now there is the ability to trigger a new workflow via the GitHub Actions interface?

Changes like CNB version bumps, lifecycle version increases will trigger a build as part of the PR merging.

The only things that will actually get updated on cron jobs are components that are unpinned, so:
- base image updates (eg after the stack image release, once every two weeks)
- classic buildpack changes (which will get pulled in automatically by cnb-shim from the stable release on buildpack registry)

Perhaps a first step might be to avoid the cron firing over the weekend (or even a Friday)?

Though I suppose this would also delay things like any buildpack registry rollbacks propagating into the builder too.

Anyway, I guess something for another PR :-)

Reply:
> Perhaps a first step might be to avoid the cron firing over the weekend (or even a Friday)?

Seems fine to me. I'll add it.

> Though I suppose this would also delay things like any buildpack registry rollbacks propagating into the builder too.

I think in this scenario, I'd prefer to trigger the action manually rather than wait for the scheduled run anyway, regardless of the frequency.